tsl/src/continuous_aggs/create.c

```diff
@@ -2284,13 +2423,30 @@ finalizequery_get_select_query(FinalizeQueryInfo *inp, List *matcollist,
 
 	CAGG_MAKEQUERY(final_selquery, inp->final_userquery);
 	final_selquery->hasAggs = !inp->finalized;
-	final_selquery->rtable = inp->final_userquery->rtable; /* fixed up above */
-	/* fixup from list. No quals on original table should be
-	 * present here - they should be on the query that populates the mattable (partial_selquery)
+	if (list_length(inp->final_userquery->jointree->fromlist) > 1)
```

Comment:
```suggestion
	if (list_length(inp->final_userquery->jointree->fromlist) >= CONTINUOUS_AGG_MAX_JOIN_RELATIONS)
```
Shouldn't be this check in that way???